userspace/libsinsp/events/sinsp_events.cpp

```diff
@@ -72,28 +73,23 @@ bool libsinsp::events::is_plugin_event(ppm_event_code event_type)
 	return (category & EC_PLUGIN);
 }
 
-std::unordered_set<std::string> libsinsp::events::event_set_to_names(const libsinsp::events::set<ppm_event_code>& events_set)
+std::unordered_set<std::string> libsinsp::events::event_set_to_names(const libsinsp::events::set<ppm_event_code>& events_set, bool resolve_generic)
 {
 	std::unordered_set<std::string> events_names_set;
-	for (const auto& val : events_set)
+	for (const auto& ev : events_set)
```

Comment:
```suggestion
	for (const auto& ev : events_set)
	{
		if (!libsinsp::events::is_generic(val))
		{
			events_names_set.insert(scap_get_event_info_table()[val].name);
		}
		else if(resolve_generic)
		{
			// Skip unknown
			for (uint32_t i = 1; i < PPM_SC_MAX; i++)
			{
				auto single_ev_set = libsinsp::events::set<ppm_sc_code>();
				single_ev_set.insert((ppm_sc_code)i);
				const auto evts = sc_set_to_event_set(single_ev_set);
				if (evts.contains(val))
				{
					events_names_set.insert(scap_get_syscall_info_table()[i].name);
				}
			}
		}
	}
```

what about keeping it exactly as it is but just using the new flag in the `if`?

Reply:
Preference gravitates towards re-using existing APIs when possible in a cleaner way than introducing yet more very hard to audit loops. Is there a significant advantage of the old method that I could be missing?

Reply:
no difference, TBH the new implementation seems more difficult to audit for my personal taste but no issue at all :)